src/BLEPeripheral.cpp

```diff
@@ -69,39 +69,54 @@ BLEPeripheral::~BLEPeripheral() {
 }
 
 void BLEPeripheral::begin() {
-  unsigned char advertisementDataType = 0;
   unsigned char scanDataType = 0;
 
-  unsigned char advertisementDataLength = 0;
+  unsigned char advertisementDataSize = 0;
   unsigned char scanDataLength = 0;
 
-  unsigned char advertisementData[BLE_ADVERTISEMENT_DATA_MAX_VALUE_LENGTH];
+  BLEAdvertisementData advertisementData[3];
```

Comment:
So, this will using about 42 more bytes of RAM on nRF8001 boards, which I think is ok. Not sure if there's a way to reduce it.